### PR TITLE
sdsl-lite: import from brewsci/science

### PR DESCRIPTION
**Pull request by suujia**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by sjackman** [comment]
```
Error: No available formula with the name "libdivsufsort"
```
You'll need to first migrate `libdivsufsort`, which this formula depends on.

**Comment by suujia** [comment]
Ok sounds good. This is taken from sdsl-lite.rb: 

```
 # this library is now part of SDSL - should remove the formula?
  conflicts_with "libdivsufsort"    
```

**Comment by sjackman** [comment]
Oh, hah. In that case, do not migrate `libdivsufsort`, and instead remove
```ruby
  conflicts_with "libdivsufsort"    
```

**Comment by suujia** [comment]
I have updated it. Can we run the tests again?

**Comment by sjackman** [comment]
```
brewsci/bio/sdsl-lite:
  * New formulae should not have a HEAD spec
```
Although I like having the `head` URL, `brew audit` complains when it exists in a new formula, which prevents bottles being created, and we don't currently have a way to disable this test. So, for now, please remove the `head` line. We can add it back in after merging this PR.

**Comment by sjackman** [comment]
```
brewsci/bio/sdsl-lite:
  * C: 13: col 1: Trailing whitespace detected.
```

**Comment by sjackman** [comment]
CircleCI on Linux
```
[ 87%] Building CXX object lib/CMakeFiles/sdsl.dir/structure_tree.cpp.o
g++-5: internal compiler error: Killed (program cc1plus)
```
https://circleci.com/gh/brewsci/homebrew-bio/2501

This error happens when it uses too much memory on CircleCI on Linux, which has only 4 GB of memory. For the fix, see for example
https://github.com/brewsci/homebrew-bio/blob/c76935a3896354ab70354f15966d50175c84cc95/Formula/fwdpp.rb#L25-L26

Try first 16, if that fails, try 8, then 4, 2, 1, until you find one that works.

**Comment by jonchang** [comment]
Looks like 8 is the magic number.

```
  * C: 22: col 1: Trailing whitespace detected.
  * C: 23: col 6: Incorrect indentation detected (column 5 instead of 4).
```

**Comment by sjackman** [comment]
On both CircleCI for Linux and TravisCI.
```
==> brew test brewsci/bio/sdsl-lite --verbose
Testing brewsci/bio/sdsl-lite
Error: brewsci/bio/sdsl-lite: failed
The selected compiler doesn't support C++11: c_compiler
```
https://travis-ci.org/brewsci/homebrew-bio/builds/377023241#L2533

This error is odd. GCC 5 definitely supports C++11.

File: Formula/sdsl-lite.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  depends_on "cmake" => :build
+
+  needs :cxx11
+
+  def install
+    ENV.cxx11
+    system "./install.sh", prefix
+    pkgshare.install "examples", "extras", "tutorial"
+    # Reduce memory usage for Circle CI.
+    ENV["MAKEFLAGS"] = "-j8" if ENV["CIRCLECI"]
```
Comment (sjackman) [inline Formula/sdsl-lite.rb]: Setting `MAKEFLAGS` needs to happen before `install.sh` to have an effect.

**Comment by sjackman** [comment]
```
The selected compiler doesn't support C++11: c_compiler
```

This error will be resolved by PR https://github.com/Linuxbrew/brew/pull/705 once it's merged.

**Comment by sjackman** [comment]
Merged PR https://github.com/Linuxbrew/brew/pull/705

**Comment by suujia** [comment]
Rebuild:  Circle CI passes, travis error: 
``` 
The job exceeded the maximum time limit for jobs, and has been terminated.
```

**Comment by sjackman** [comment]
Merged. Thanks, Susie!

**Pull request closed without merging**